### PR TITLE
Roots are added out of order to the accounts index

### PR DESCRIPTION
**Pull request by aeyakovenko**
#### Problem

Bank adds roots out of order.

#### Summary of Changes

Add roots in order.  Assert roots are added in order.

Fixes #

### CODE REVIEW

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -293,7 +293,7 @@ impl Bank {
         *self.parent.write().unwrap() = None;
 
         let squash_accounts_start = Instant::now();
-        for p in &parents {
+        for p in parents.iter().rev() {
```
Comment (rob-solana) [inline runtime/src/bank.rs]: 😮

Comment (aeyakovenko) [reply runtime/src/bank.rs]: @rob-solana i am not sure if getting the slot values out of self.ancestors and sorting them will be better.

Comment (rob-solana) [reply runtime/src/bank.rs]: I'm worried that they might not be the same...

squash() doesn't change bank.ancestors, does it?

Comment (aeyakovenko) [reply runtime/src/bank.rs]: ah, it might add extra roots?

Comment (rob-solana) [reply runtime/src/bank.rs]: dunno.  I'm trying to understand how ancestors is used.  a hashmap of slot->depth that's passed all the way down to accounts_index (which is **really** opaque) is hard for me to prove to myself that it's correct

Comment (aeyakovenko) [reply runtime/src/bank.rs]: Write some tests :).  The roots table in the index does the magic part.

**Pull request merged**